openstack/compute/v2/extensions/instanceactions/results.go

```diff
@@ -58,13 +58,77 @@ func ExtractInstanceActions(r pagination.Page) ([]InstanceAction, error) {
 	return resp, err
 }
 
+// Event represents an event of instance action.
+type Event struct {
+	Event      string    `json:"event"`
+	Host       string    `json:"host"`
+	HostID     string    `json:"hostId"`
+	Result     string    `json:"result"`
+	Traceback  string    `json:"traceback"`
+	StartTime  time.Time `json:"-"`
+	FinishTime time.Time `json:"-"`
+}
+
+// UnmarshalJSON converts our JSON API response into our instance action struct
+func (e *Event) UnmarshalJSON(b []byte) error {
+	type tmp Event
+	var s struct {
+		tmp
+		StartTime  gophercloud.JSONRFC3339MilliNoZ `json:"start_time"`
+		FinishTime gophercloud.JSONRFC3339MilliNoZ `json:"finish_time"`
+	}
+	err := json.Unmarshal(b, &s)
+	if err != nil {
+		return err
+	}
+	*e = Event(s.tmp)
+
+	e.StartTime = time.Time(s.StartTime)
+	e.FinishTime = time.Time(s.FinishTime)
+
+	return err
+}
+
+// InstanceActionDetail gives more details on instance action.
+type InstanceActionDetail struct {
+	Action       string    `json:"action"`
```

Comment:
tried with nested type InstanceAction but the func extractIntoPtr didn't work, So I copy paste all field :/